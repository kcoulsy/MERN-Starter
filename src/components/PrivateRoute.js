import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';

function PrivateRoute({ isLoggedIn, children, ...rest }) {
    return (
        <Route
            {...rest}
            render={(props) => {
                if (!isLoggedIn) {
                    return <Redirect to={{ pathname: '/login', state: { from: props.location } }} />;
                }

                return children;
            }}
        />
    );
}

const mapStateToProps = (state) => {
    return {
        isLoggedIn: typeof state.auth.token === 'string' && state.auth.token.length,
    };
};

export default connect(mapStateToProps)(PrivateRoute);
