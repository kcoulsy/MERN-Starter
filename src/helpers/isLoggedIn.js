export default (token) => typeof token === 'string' && token.length > 0;
