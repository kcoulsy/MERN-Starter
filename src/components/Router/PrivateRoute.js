import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import isLoggedIn from '../../helpers/isLoggedIn';

function PrivateRoute({ isLoggedIn, children, ...rest }) {
    return (
        <Route
            {...rest}
            render={(props) => {
                if (!isLoggedIn) {
                    return <Redirect to={{ pathname: '/login', state: { from: props.location } }} />;
                }

                return children;
            }}
        />
    );
}

const mapStateToProps = (state) => {
    return {
        isLoggedIn: isLoggedIn(state.auth.token),
    };
};

export default connect(mapStateToProps)(PrivateRoute);
