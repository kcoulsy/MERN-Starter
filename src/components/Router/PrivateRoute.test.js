import React from 'react';
import { shallow } from 'enzyme';

import { PrivateRoute, privateRouteRender, mapStateToProps } from './PrivateRoute';
import { shallowToJson } from 'enzyme-to-json';
import isLoggedIn from '../../helpers/isLoggedIn';

describe('render PrivateRoute component', () => {
    it('should render PrivateRoute component', () => {
        const wrapper = shallow(<PrivateRoute />);

        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should handle PrivateRouteRender when logged in', () => {
        const child = <p>JSX Child</p>;
        const isLoggedIn = true;
        const wrapper = shallow(privateRouteRender(isLoggedIn, child)({ location: jest.fn() }));

        expect(wrapper.contains(child)).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });

    it('should handle PrivateRouteRender when not logged in', () => {
        const child = <p>JSX Child</p>;
        const isLoggedIn = false;
        const wrapper = shallow(privateRouteRender(isLoggedIn, child)({ location: jest.fn() }));

        expect(wrapper.contains(child)).toBe(false);
        expect(wrapper).toMatchSnapshot();
    });

    it('should return correct props from store', () => {
        const testStore = {
            auth: {
                token: 'testtoken',
            },
        };
        expect(mapStateToProps(testStore)).toEqual({
            isLoggedIn: isLoggedIn(testStore.auth.token),
        });
    });
});
