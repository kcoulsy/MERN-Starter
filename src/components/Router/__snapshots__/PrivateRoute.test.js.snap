// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render PrivateRoute component should handle PrivateRouteRender when logged in 1`] = `
<p>
  JSX Child
</p>
`;

exports[`render PrivateRoute component should handle PrivateRouteRender when not logged in 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`render PrivateRoute component should render PrivateRoute component 1`] = `
<Route
  render={[Function]}
/>
`;
